flow/Platform.cpp

```diff
@@ -1851,7 +1851,12 @@ std::string parentDirectory( std::string const& filename ) {
 			.GetLastError();
 		throw platform_error();
 	}
-	return abs.substr(0, sep);
+
+	while (abs.size() && (abs.back() == '/' || abs.back() == CANONICAL_PATH_SEPARATOR)) {
+		abs.resize(abs.size() - 1);
+	}
+
+	return abs.substr(0, sep+1);
```

Comment:
The while loop shortens abs to have no trailing separators, but sep points to the last separator in the string, which could be one of the characters removed by the while loop, so then the returned value could be invalid.

Reply:
After much discussion, we've determined that amazingly this code actually works (we think).  Substr has a safety check which will reduce lengths that are too large to just the remainder of the string, and abspath() will return something that does not contain a trailing separator unless the string only contains separators (referring to root).